src/components/utils/test/json_utils_test.cc

```diff
@@ -0,0 +1,1139 @@
+      ASSERT_EQ(ref.Size(), index_ref.Size());
+      index++;
+    }
+  }
+  ASSERT_EQ(kTestJsonArraySize, index);
+}
+
+TEST(TestJsonVaue, ArrayIterateByIndex) {
+  using namespace utils::json;
+  JsonValue::ParseResult parse_result = JsonValue::Parse(kTestJsonArrayString);
+  ASSERT_TRUE(parse_result.second);
+
+  JsonValue& root_json = parse_result.first;
+  ASSERT_EQ(ValueType::ARRAY_VALUE, root_json.Type());
+  const JsonValue::ArrayIndex size = root_json.Size();
+  ASSERT_EQ(kTestJsonArraySize, size);
+  JsonValue::ArrayIndex index = 0;
+  for (; index < size; ++index) {
+    JsonValueRef index_ref = root_json[index];
+    if (index_ref.IsUInt()) {
+      ASSERT_EQ(5, index_ref.AsInt());
+      index++;
+    } else if (index_ref.IsObject()) {
+      JsonValueRef ref = root_json[0u];
+      ASSERT_EQ(ref.Size(), index_ref.Size());
+      index++;
+    }
+  }
+  ASSERT_EQ(kTestJsonArraySize, index);
+}
+
+TEST(TestJsonVaue, SubArrayIterator) {
+  using namespace utils::json;
+  JsonValue::ParseResult parse_result = JsonValue::Parse(kTestJsonObjectString);
+  ASSERT_TRUE(parse_result.second);
+
+  JsonValue& root_json = parse_result.first;
+  ASSERT_EQ(ValueType::OBJECT_VALUE, root_json.Type());
+  ASSERT_EQ(kTestJsonObjectSize, root_json.Size());
+  JsonValueRef sub_array = root_json["ArrVal"];
+  ASSERT_TRUE(sub_array.IsArray());
+  ASSERT_EQ(15u, sub_array.Size());
+  uint32_t expected_value = 3;
+  for (JsonValue::iterator itr = sub_array.begin(), end = sub_array.end();
+       itr != end;
+       ++itr) {
+    ASSERT_EQ(expected_value++, (*itr).AsUInt());
+  }
+}
+
+TEST(TestJsonVaue, SubArrayConstIterator) {
+  using namespace utils::json;
+  JsonValue::ParseResult parse_result = JsonValue::Parse(kTestJsonObjectString);
+  ASSERT_TRUE(parse_result.second);
+
+  const JsonValueRef& root_json = parse_result.first;
+  ASSERT_EQ(ValueType::OBJECT_VALUE, root_json.Type());
+  ASSERT_EQ(kTestJsonObjectSize, root_json.Size());
+  const JsonValueRef& sub_array = root_json["ArrVal"];
+  ASSERT_TRUE(sub_array.IsArray());
+  ASSERT_EQ(15u, sub_array.Size());
+  uint32_t expected_value = 3;
+  for (JsonValue::const_iterator itr = sub_array.begin(), end = sub_array.end();
+       itr != end;
+       ++itr) {
+    ASSERT_EQ(expected_value++, (*itr).AsUInt());
+  }
+}
+
+TEST(TestJsonVaue, EmptyObjectIterator) {
+  using namespace utils::json;
+  JsonValue::ParseResult parse_result = JsonValue::Parse(kTestJsonObjectString);
+  ASSERT_TRUE(parse_result.second);
+
+  JsonValueRef root_json = parse_result.first;
+  ASSERT_EQ(ValueType::OBJECT_VALUE, root_json.Type());
+  ASSERT_EQ(kTestJsonObjectSize, root_json.Size());
+  JsonValueRef obj_val = root_json["ObjVal"];
+  ASSERT_EQ(ValueType::OBJECT_VALUE, obj_val.Type());
+  ASSERT_GT(obj_val.Size(), 0u);
+  obj_val.Clear();
+  ASSERT_EQ(0u, obj_val.Size());
+  uint32_t index = 0;
+  for (JsonValue::iterator itr = obj_val.begin(), end = obj_val.end();
+       itr != end;
+       ++itr) {
+    index++;
```

Comment:
@OHerasym Is there a method in `JsonValue` which returns its size to avoid this loop here and in other places?